fixieai/agents-ts/src/embed.ts

```diff
@@ -0,0 +1,111 @@
+
+function isURL(str: string) {
+  try {
+    new URL(str);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+function uriFromBase64(base64Data: string): string {
+  return `data:base64,${base64Data}`;
+}
+
+export function base64FromUri(uri: string): string {
+  const [, base64Data] = uri.split('data:base64,');
+  return base64Data;
+}
+
+export interface SerializedEmbed extends Pick<Embed, 'uri'> {
+  content_type: string;
+}
+
+/**
+ * A data object attached to a Message. This is useful when you want to pass data between agents. For
+ * instance, you might want to generate and pass an image. Or, if you have text, and you want to pass that text
+ * atomically (so it doesn't get munged by other LLMs), you can pass it as an embed.
+ */
+export class Embed {
+  public constructor(
+    /**
+     * The MIME content type of the object, e.g., "image/png" or "application/json".
+     *
+     * https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+     */
+    public readonly contentType: string,
+    /**
+     * The URI for this embed.
+     *
+     * This URI could either point to an external resource, or be a base64-encoded data URI.
+     */
+    public readonly uri: string,
+  ) {}
+
+  /**
+   * Format this Embed for sending to the Fixie API.
+   */
+  serialize(): SerializedEmbed {
+    return {
+      content_type: this.contentType,
+      uri: this.uri,
+    };
+  }
+
+  /**
+   * Get the embed's contents as text. If the embed was created from a URI pointing to an external resource, this method will fetch the URI. If the embed was created from base64 data, this method will decode the base64 data.
+   *
+   * @returns this embed's data as a UTF-8 string.
+   */
+  async loadDataAsText() {
+    return (await this.getDataAsBinary()).toString('utf-8');
+  }
+
+  /**
+   * Get the embed's contents as text. If the embed was created from a URI pointing to an external resource, this method will fetch the URI. If the embed was created from base64 data, this method will decode the base64 data.
+   *
+   * @returns this embed's data as a [Buffer](https://nodejs.org/docs/latest/api/buffer.html).
+   */
+  async getDataAsBinary() {
+    if (this.uri.startsWith('data:base64,')) {
+      return Buffer.from(base64FromUri(this.uri), 'base64');
+    }
+
+    const response = await got(this.uri, {
+      responseType: 'buffer',
+    });
+
+    if (response.statusCode !== 200) {
+      throw new Error(
+        `Got status code ${response.statusCode} when fetching ${this.uri}`,
+      );
+    }
+
+    return response.body;
+  }
+
+  /**
+   * Create an embed from base64 data.
+   *
+   * @param contentType The MIME content type of the object, e.g., "image/png" or "application/json" (https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types)
+   * @param textData The base64-encoded data for this embed.
+   */
+  static fromBase64(contentType: string, textData: string): Embed {
```

Comment:
Seems like this should be `fromText` or `fromString` (Lines 102-103 seem to do the base64 conversion)